### PR TITLE
HBASE-27419 Update to hbase-thirdparty 4.1.2

### PR DESCRIPTION
**Pull request by Apache9**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 56s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 18s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 22s |  master passed  |
| +1 :green_heart: |  compile  |   6m 26s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 41s |  branch has no errors when running spotless:check.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m  6s |  the patch passed  |
| +1 :green_heart: |  compile  |   6m 12s |  the patch passed  |
| -0 :warning: |  javac  |   6m 12s |  root generated 1 new + 702 unchanged - 1 fixed = 703 total (was 703)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  hadoopcheck  |   7m 56s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.4.  |
| +1 :green_heart: |  spotless  |   0m 39s |  patch has no errors when running spotless:check.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 30s |  The patch does not generate ASF License warnings.  |
|  |   |  33m 38s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4818/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4818 |
| Optional Tests | dupname asflicense javac hadoopcheck spotless xml compile |
| uname | Linux 4f21b71a6f29 5.4.0-124-generic #140-Ubuntu SMP Thu Aug 4 02:23:37 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 46d37a70d6 |
| Default Java | Temurin-1.8.0_345-b01 |
| javac | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4818/1/artifact/yetus-general-check/output/diff-compile-javac-root.txt |
| Max. process+thread count | 138 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-examples . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4818/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-protocol-shaded/pom.xml [hunk 1]
```diff
@@ -34,7 +34,7 @@
     <!--Version of protobuf that hbase uses internally (we shade our pb)
          Must match what is out in hbase-thirdparty include.
     -->
-    <internal.protobuf.version>3.21.1</internal.protobuf.version>
+    <internal.protobuf.version>3.21.7</internal.protobuf.version>
```
Comment (apurtell) [inline hbase-protocol-shaded/pom.xml:37]: Ditto

File: hbase-examples/pom.xml [hunk 2]
```diff
@@ -33,7 +33,7 @@
     <!--Version of protobuf that hbase uses internally (we shade our pb)
          Must match what is out in hbase-thirdparty include.
     -->
-    <internal.protobuf.version>3.17.3</internal.protobuf.version>
+    <internal.protobuf.version>3.21.7</internal.protobuf.version>
```
Comment (apurtell) [inline hbase-examples/pom.xml:36]: Maybe there is a way to tie this setting to hbase-thirdparty somehow. Have hbase-thirdparty export a version property in its POM that we can use here.

File: pom.xml [hunk 3]
```diff
@@ -1734,6 +1734,12 @@
       <scope>test</scope>
     </dependency>
   </dependencies>
+  <repositories>
+    <repository>
+      <id>staging</id>
+      <url>https://repository.apache.org/content/repositories/orgapachehbase-1496/</url>
```
Comment (apurtell) [inline pom.xml:1740]: Should we just wait for a release? We don't need 'staging' here then.

**Review by apurtell — APPROVED** [review]

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  4s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 19s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 49s |  master passed  |
| +1 :green_heart: |  compile  |   1m 58s |  master passed  |
| +1 :green_heart: |  shadedjars  |   3m 49s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   2m 23s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 18s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 29s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 56s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 56s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   3m 48s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   2m 24s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 273m 33s |  root in the patch passed.  |
|  |   | 299m 37s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4818/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4818 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux b3be86ebce9b 5.4.0-122-generic #138-Ubuntu SMP Wed Jun 22 15:00:31 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 46d37a70d6 |
| Default Java | Eclipse Adoptium-11.0.16.1+1 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4818/1/testReport/ |
| Max. process+thread count | 4978 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-examples . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4818/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   3m 28s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 26s |  master passed  |
| +1 :green_heart: |  compile  |   1m 30s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 11s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 41s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 30s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m  6s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 29s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 29s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m  6s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 42s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 401m 59s |  root in the patch passed.  |
|  |   | 428m 48s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4818/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4818 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 73dc7656bac2 5.4.0-1071-aws #76~18.04.1-Ubuntu SMP Mon Mar 28 17:49:57 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 46d37a70d6 |
| Default Java | Temurin-1.8.0_345-b01 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4818/1/testReport/ |
| Max. process+thread count | 4867 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-examples . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4818/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-protocol-shaded/pom.xml (hunk 1 above)
Comment (Apache9) [reply hbase-protocol-shaded/pom.xml:37]: Will open a new issue if I find a way to do this.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 19s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 19s |  master passed  |
| +1 :green_heart: |  compile  |   6m  2s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 37s |  branch has no errors when running spotless:check.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m  3s |  the patch passed  |
| +1 :green_heart: |  compile  |   5m 56s |  the patch passed  |
| -0 :warning: |  javac  |   5m 56s |  root generated 12 new + 691 unchanged - 12 fixed = 703 total (was 703)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  hadoopcheck  |   8m  8s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.4.  |
| +1 :green_heart: |  spotless  |   0m 36s |  patch has no errors when running spotless:check.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 24s |  The patch does not generate ASF License warnings.  |
|  |   |  32m 45s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4818/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4818 |
| Optional Tests | dupname asflicense javac hadoopcheck spotless xml compile |
| uname | Linux d53c3478ce90 5.4.0-1081-aws #88~18.04.1-Ubuntu SMP Thu Jun 23 16:29:17 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 46d37a70d6 |
| Default Java | Temurin-1.8.0_345-b01 |
| javac | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4818/2/artifact/yetus-general-check/output/diff-compile-javac-root.txt |
| Max. process+thread count | 140 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-examples . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4818/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  4s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 19s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 17s |  master passed  |
| +1 :green_heart: |  compile  |   2m 33s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 35s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   2m 53s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 18s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m  7s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 16s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 16s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 36s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   3m  0s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 256m 53s |  root in the patch failed.  |
|  |   | 287m 23s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4818/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4818 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 3c4a7625294a 5.4.0-122-generic #138-Ubuntu SMP Wed Jun 22 15:00:31 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 46d37a70d6 |
| Default Java | Eclipse Adoptium-11.0.16.1+1 |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4818/2/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4818/2/testReport/ |
| Max. process+thread count | 2408 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-examples . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4818/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 50s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 19s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 39s |  master passed  |
| +1 :green_heart: |  compile  |   1m 49s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 37s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   2m  6s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 37s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 52s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 52s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 18s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   2m 11s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 435m 19s |  root in the patch passed.  |
|  |   | 462m 27s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4818/2/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4818 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux a835ac402e83 5.4.0-1081-aws #88~18.04.1-Ubuntu SMP Thu Jun 23 16:29:17 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 46d37a70d6 |
| Default Java | Temurin-1.8.0_345-b01 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4818/2/testReport/ |
| Max. process+thread count | 4786 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-examples . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4818/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**